cytodynamics-nucleus/src/main/java/com/linkedin/cytodynamics/nucleus/IsolatingClassLoader.java

```diff
@@ -113,7 +114,9 @@
         return cl;
       } else {
         // got through parent and fallback delegates but could not find the class
-        throw new ClassNotFoundException("Could not load class for name " + name);
+        throw new CytodynamicsClassNotFoundException(String.format(
+            "Could not fully load class for name %s. It is possible that the immediate class is found, but a class that "
+                + "it depends on cannot be found", name));
```

Comment:
Would it be possible to root cause the actual class that was missing?

Reply:
This would indeed be helpful, but unfortunately, it doesn't seem trivial to add the root cause. I'll make a note to do some more investigation into this, but it is out of scope of this PR.

Reply:
sounds good.